docs/deepsemgrep.md

```diff
@@ -0,0 +1,417 @@
+---
+slug: deepsemgrep
+description: "This document provides an overview of DeepSemgrep use cases, such as use of DeepSemgrep in type inferences, class inheritance, constant propagation, taint analysis."
+---
+
+# DeepSemgrep
+
+## Introduction
+
+Improve your scan results for entire codebases with interfile coding paradigms using DeepSemgrep instead of Semgrep's regular intrafile (within-a-single-file) approach. In most object-oriented programming styles, it is expected that classes are in different files. DeepSemgrep empowers you to easily scan whole repositories with classes in different files to find vulnerabilities in your code. DeepSemgrep is a proprietary extension of free and open source Semgrep which leverages global analysis tools, and uses the same rules as Semgrep.
+
+This document demonstrates the utility of DeepSemgrep through use cases, guiding you through examples of Semgrep rules that you can use to search for particular patterns in code. Both Semgrep rules and code on which rules are tested are marked as code snippets and introduced always as a rule, code, or file used to illustrate the capabilities of DeepSemgrep.
+
+## Obtaining DeepSemgrep
+
+To get DeepSemgrep, follow these steps:
+1. Submit your email on the following website [DeepSemgrep beta form](https://semgrep.dev/deep-semgrep-beta).
+2. Semgrep team contacts you at your email address, follow the steps and instructions in the email.
+3. Start using DeepSemgrep by using the following command:
+    ```sh
+    semgrep login
+    ```
+4. Use a link that Semgrep prints to the command-line.
+5. Use the following commands:
+    ```sh
+    semgrep install-deep-semgrep
+    ```
+
+    And then:
+    ```sh
+    semgrep --deep --config=myrule.yaml
+    ```
```

Comment:
Could these two steps be combined into one code block?

Reply:
Usually, I understand that people can read it, but it requires a certain cognitive investment, which we are usually trying to reduce in docs when possible. I rather divided it into separate steps:

```
5. Use the following commands:
    ```sh
    semgrep install-deep-semgrep
    ```
6. To test DeepSemgrep, use the following command:
    ```sh
    semgrep --deep --config=myrule.yaml
    ```
```